packages/mimir/src/rules/no-useless-spread.ts

```diff
@@ -15,20 +24,51 @@ export class Rule extends AbstractRule {
                 match.index,
             )!;
             if (
-                (
-                    isSpreadElement(node) && node.expression.kind === ts.SyntaxKind.ArrayLiteralExpression ||
-                    isSpreadAssignment(node) && node.expression.kind === ts.SyntaxKind.ObjectLiteralExpression
-                ) &&
+                ((isSpreadElement(node) &&
+                    isArrayLiteralExpression(node.expression) &&
+                    (!node.expression.elements.some(isOmittedExpression) || isReassignmentTarget(node))) ||
+                    (isSpreadAssignment(node) && isObjectLiteralExpression(node.expression))) &&
                 node.expression.pos - 3 === match.index
             )
                 this.addFailureAtNode(node, FAILURE_STRING, removeUselessSpread(node));
         }
     }
 }
 
-function removeUselessSpread(node: ts.SpreadElement | ts.SpreadAssignment): Replacement[] {
-    return [
+function removeUselessSpread(node: ts.SpreadElement | ts.SpreadAssignment): Replacement | Replacement[] {
+    /* Handles edge cases of spread on empty array literals */
+    if (
+        isSpreadElement(node) &&
+        isArrayLiteralExpression(node.expression) &&
+        node.expression.elements.length === 0 &&
+        node.parent &&
+        isArrayLiteralExpression(node.parent)
```

Comment:
this should not be specific to array literals. this part of the condition should be removed to handle all possible parents of SpreadElement:

```ts
console.log('foo', ...[],);
// currently fixed to
console.log('foo',,);
// should be fixed to
console.log('foo',);

// same with constructor calls
new Foo('foo', ...[],);
```

Reply:
Yeah you're right and the TS compiler was telling me as much... What's the cleanest or most efficient way to grab the next comma if it exists?